llvm/lib/DebugInfo/GSYM/CallSiteInfo.cpp

```diff
@@ -0,0 +1,293 @@
+    std::string FuncName = stringFromOffset(Function.Name).str();
+    // If the function name is already in the map, don't update it. This way we
+    // preferentially use the first encountered function. Since symbols are
+    // loaded from dSYM first, we end up preferring keeping track of symbols
+    // from dSYM rather than from the symbol table - which is what we want to
+    // do.
+    if (FuncMap.count(FuncName))
+      return;
+    FuncMap[FuncName] = &Function;
+  };
+  for (auto &Func : Funcs) {
+    insertFunc(Func);
+    if (Func.MergedFunctions.has_value())
+      for (auto &MFunc : Func.MergedFunctions->MergedFunctions)
+        insertFunc(MFunc);
+  }
+  return FuncMap;
+}
+
+llvm::Error CallSiteInfoLoader::processYAMLFunctions(
+    const llvm::yaml::FunctionsYAML &functionsYAML,
+    std::unordered_map<std::string, FunctionInfo *> &FuncMap,
+    StringRef YAMLFile) {
+  // For each function in the YAML file
+  for (const auto &FuncYAML : functionsYAML.functions) {
+    auto it = FuncMap.find(FuncYAML.name);
+    if (it == FuncMap.end()) {
+      return llvm::createStringError(
+          std::errc::invalid_argument,
+          "Can't find function '%s' specified in callsite YAML\n",
+          FuncYAML.name.c_str());
+    }
+    FunctionInfo *FuncInfo = it->second;
+    // Create a CallSiteInfoCollection if not already present
+    if (!FuncInfo->CallSites)
+      FuncInfo->CallSites = CallSiteInfoCollection();
+    for (const auto &CallSiteYAML : FuncYAML.callsites) {
+      CallSiteInfo CSI;
+      // Since YAML has specifies relative return offsets, add the function
+      // start address to make the offset absolute.
+      CSI.ReturnAddress = FuncInfo->Range.start() + CallSiteYAML.return_offset;
+      for (const auto &regex : CallSiteYAML.match_regex) {
+        CSI.MatchRegex.push_back(offsetFromString(regex));
+      }
+      // Initialize flags to None
+      CSI.Flags = CallSiteInfo::None;
+      // Parse flags and combine them
+      for (const auto &FlagStr : CallSiteYAML.flags) {
+        if (FlagStr == "InternalCall") {
+          CSI.Flags |= static_cast<uint8_t>(CallSiteInfo::InternalCall);
+        } else if (FlagStr == "ExternalCall") {
+          CSI.Flags |= static_cast<uint8_t>(CallSiteInfo::ExternalCall);
+        } else {
+          return llvm::createStringError(std::errc::invalid_argument,
+                                         "Unknown flag in callsite YAML: %s\n",
+                                         FlagStr.c_str());
+        }
+      }
+      FuncInfo->CallSites->CallSites.push_back(CSI);
+    }
+  }
+  return llvm::Error::success();
+}
+
+raw_ostream &llvm::gsym::operator<<(raw_ostream &OS, const CallSiteInfo &CSI) {
+  OS << "  Return=" << HEX64(CSI.ReturnAddress);
+  OS << "  Flags=" << HEX8(CSI.Flags);
+
+  OS << "  RegEx=";
+  for (uint32_t i = 0; i < CSI.MatchRegex.size(); ++i) {
+    if (i > 0)
+      OS << ",";
+    OS << CSI.MatchRegex[i];
```

Comment:
Looks like it prints out offset to the string table. It could be more useful to print the actual content, but not sure how we could do here.